src/shogun/classifier/AveragedPerceptron.cpp

```diff
@@ -40,22 +40,10 @@ void AveragedPerceptron::init()
 	    ParameterProperties::MODEL)
 }
 
-void AveragedPerceptron::init_model(std::shared_ptr<Features> data)
+void AveragedPerceptron::init_model(const std::shared_ptr<Features>& data)
```

Comment:
yes, this is really something we should pay attention on doing ... as then we might be able to make features/labels const in `train`